### PR TITLE
feat: add oer info page (to be done)

### LINKED ISSUE #567: Page about OER
[OER erklärt - die Grundlagen](https://www.bpb.de/mediathek/video/234994/oer-erklaert-die-grundlagen/)
[OER erklärt - über die Qualität der Materialien](https://www.bpb.de/mediathek/video/234998/oer-erklaert-ueber-die-qualitaet-der-materialien/)

Link to Youtube:
https://www.youtube.com/watch?v=DfJWqoEc6BI
https://www.youtube.com/watch?v=xm0eLHBzO-Y

Options that we can add on the embedded player to improve the localization:
- `cc_lang_pref` and `hl` can be set to the user language for better integration
- set `rel` to 0 to limit what videos are displayed after (uses videos from the same channel)

Maybe investigate which web players could be used for showing captions
Do we even need captions ?

### PR DESCRIPTION
**Pull request by pyphilia**
close #567 

Screenshot 2024-04-29 at 15 38 06
Screenshot 2024-04-29 at 15 38 20
Screenshot 2024-04-29 at 15 38 34
Screenshot 2024-04-29 at 15 38 42

### CODE REVIEW

File: src/components/collection/oer/VideoWithCC.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  return (
+    <Stack
+      p={3}
+      style={{
+        borderRadius: 20,
+        border: '1px solid lightgrey',
+        textAlign: 'center',
+        background: 'white',
+        width: 'fit-content',
+        margin: 'auto',
```
Comment (spaenleh) [inline src/components/collection/oer/VideoWithCC.tsx:37]: These props are available on the stack directly (i think)

File: src/components/collection/oer/VideoWithCC.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      }}
+      gap={3}
+      alignContent="center"
+    >
+      <Stack direction="row" justifyContent="center" alignItems="center">
```
Comment (spaenleh) [inline src/components/collection/oer/VideoWithCC.tsx:42]: You don't need a Stack for a single element.

File: src/components/collection/oer/VideoWithCC.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          width="560"
+          height="315"
+          src={url}
+          title={title}
+          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+          allowFullScreen
+        />
+      </Stack>
+      <Stack direction="row" justifyContent="center" alignItems="center">
+        <Stack>
```
Comment (spaenleh) [inline src/components/collection/oer/VideoWithCC.tsx:54]: Remove this lonely stack.

File: src/components/collection/oer/VideoWithCC.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              <ListItem>
+                <strong>
+                  {t(LIBRARY.OER_INFORMATION_VIDEO_DESCRIPTION_EDITION)}
+                </strong>
+                : {edition}
+              </ListItem>
+            )}
+          </List>
+        </Stack>
+        <Stack>
```
Comment (spaenleh) [inline src/components/collection/oer/VideoWithCC.tsx:82]: Same here, no need to wrap with a stack specifically, maybe a box if you really need.

File: src/components/common/CreativeCommons.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    allowCommercialUse: !ccLicenseAdaption?.includes('NC'),
+    allowSharing: (() => {
+      if (!ccLicenseAdaption || !ccLicenseAdaption.length) {
+        return '';
+      }
+      if (ccLicenseAdaption?.includes('SA')) {
+        return 'alike';
+      }
+      return ccLicenseAdaption?.includes('ND') ? 'no' : 'yes';
+    })(),
```
Comment (spaenleh) [inline src/components/common/CreativeCommons.tsx:35]: Define the function above and call it below ? Here it is a bit confusing that you have the declaration and call on the same place

File: src/components/common/CreativeCommons.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  id,
+  ccLicenseAdaption,
+  iconSize = 48,
+}: Props): JSX.Element => {
+  const { t } = useLibraryTranslation();
+
+  const { allowSharing, allowCommercialUse, requireAccreditation } = useMemo(
+    () => convertLicense(ccLicenseAdaption ?? ''),
+    [ccLicenseAdaption],
+  );
```
Comment (spaenleh) [inline src/components/common/CreativeCommons.tsx:55]: Is the memo needed here ? you are computing the values from the function given a prop. Maybe the function should simply be a custom hook. I don't think that there is any thing else that would re-render the component ?

File: src/components/common/TwitterButton.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    <Tooltip title={t(LIBRARY.SHARE_TWITTER_TOOLTIP)}>
+      <IconButton
+        sx={{
+          marginLeft: 1,
+          marginRight: 1,
+          border: showBorder ? `1px solid lightgrey` : 'none',
+        }}
+        color="primary"
+        onClick={shareOnTwitter}
+      >
```
Comment (spaenleh) [inline src/components/common/TwitterButton.tsx:51]: Would it make sens to factor this button out, so the common style would be defined in one place ?

File: src/components/pages/OERInformation.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
+'use client';
```
Comment (spaenleh) [inline src/components/pages/OERInformation.tsx:1]: Do you need the boundary that high ? Would it be ok to lower it ?

File: src/components/pages/OERInformation.tsx [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            />
+            <EmailButton
+              showBorder
+              iconSize="medium"
+              name={t(LIBRARY.OER_INFORMATION_PAGE_TITLE)}
+              description={t(LIBRARY.OER_INFORMATION_PAGE_DESCRIPTION)}
+            />
+          </Box>
+          <Divider variant="middle" />
+        </Box>
```
Comment (spaenleh) [inline src/components/pages/OERInformation.tsx:65]: Maybe extract this into a local component that you add in this file.

File: src/components/pages/OERInformation.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+            <VideoWithCC
+              url="https://www.youtube.com/embed/DfJWqoEc6BI?si=fuhfrbKpqXUN4Vcx"
+              title={t(LIBRARY.OER_INFORMATION_BASICS_TITLE)}
+              production="Kooperative Berlin"
+              duration="3 Min."
+              edition="Bundeszentrale für politische Bildung"
+            />
+          </Container>
+        </StyledBox>
```
Comment (spaenleh) [inline src/components/pages/OERInformation.tsx:88]: Same here, extract to a local component ?

**Review by spaenleh — APPROVED** [review]
Great work ! Really like the video frames you created. I propose to extract some of the OER page sections into local components to ease the reading of the main component. 

Maybe the references could be added inside the video frames so it is easy to click ? Otherwise thank you very much for the amazing work ! 💪

File: src/components/pages/OERInformation.tsx (hunk 10 above)
Comment (pyphilia) [reply src/components/pages/OERInformation.tsx:88]: I think it's good for now, not sure how this page will evolve anyway.

File: src/components/pages/OERInformation.tsx (hunk 8 above)
Comment (pyphilia) [reply src/components/pages/OERInformation.tsx:1]: would be indeed cool! But we need to use https://github.com/i18next/next-i18next
I move this to another issue.

**Pull request merged**